src/cli/rustup_mode.rs

```diff
@@ -1578,6 +1585,14 @@ fn set_profile(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
     Ok(utils::ExitCode(0))
 }
 
+fn set_auto_self_update(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
+    if self_update::NEVER_SELF_UPDATE {
+        warn!("{} is built with the no-self-update feature: setting auto-self-update will not have any effect.",cfg.rustup_dir.display());
```

Comment:
I think you need arg0 + os.path.realpath, like from proxy_mode.rs
```
        let mut args = crate::process().args_os();
        let arg0 = args.next().map(PathBuf::from);
```
     rustup_dir is just the directory that the common case install path for the binary is: we want the actual path, which for a snap, is some random directory.